src/main/java/net/neoforged/neoforge/common/util/AttributeUtil.java

```diff
@@ -0,0 +1,356 @@
+                        var fakeModif = new AttributeModifier(AttributeUtil.FAKE_MERGED_ID, sums[i], op);
+                        tooltip.accept(attr.value().toComponent(fakeModif, ctx.flag()));
+                    }
+                }
+            } else {
+                for (AttributeModifier m : modifs) {
+                    if (m.amount() != 0) {
+                        tooltip.accept(attr.value().toComponent(m, ctx.flag()));
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Adds tooltip lines for the attribute modifiers contained in a {@link PotionContents}.
+     * 
+     * @param list     The list of attribute modifiers generated by calling {@link MobEffect#createModifiers} for each mob effect instance on the potion.
+     * @param tooltips The tooltip consumer to add lines to.
+     */
+    public static void addPotionTooltip(List<Pair<Holder<Attribute>, AttributeModifier>> list, Consumer<Component> tooltips) {
+        if (!list.isEmpty()) {
+            tooltips.accept(CommonComponents.EMPTY);
+            tooltips.accept(Component.translatable("potion.whenDrank").withStyle(ChatFormatting.DARK_PURPLE));
+
+            for (Pair<Holder<Attribute>, AttributeModifier> pair : list) {
+                tooltips.accept(pair.getFirst().value().toComponent(pair.getSecond(), AttributeUtil.getTooltipFlag()));
+            }
+        }
+    }
+
+    /**
+     * Creates a sorted {@link TreeMultimap} used to ensure a stable iteration order of item attribute modifiers.
+     */
+    public static Multimap<Holder<Attribute>, AttributeModifier> sortedMap() {
+        return TreeMultimap.create(Comparator.comparing(Holder::value, Comparator.comparing(BuiltInRegistries.ATTRIBUTE::getKey, ResourceLocation::compareTo)), ATTRIBUTE_MODIFIER_COMPARATOR);
+    }
+
+    /**
+     * Returns a sorted, mutable {@link Multimap} containing all the attribute modifiers on an item stack for the given group.
+     * <p>
+     * This includes attribute modifiers from components (or default modifiers, if not present), enchantments, and the {@link ItemAttributeModifierEvent}.
+     * 
+     * @param stack The stack to query modifiers for.
+     * @param slot  The slot group to query modifiers for.
+     */
+    public static Multimap<Holder<Attribute>, AttributeModifier> getSortedModifiers(ItemStack stack, EquipmentSlotGroup slot) {
+        Multimap<Holder<Attribute>, AttributeModifier> map = AttributeUtil.sortedMap();
+        stack.forEachModifier(slot, (attr, modif) -> {
+            if (attr != null && modif != null) {
+                map.put(attr, modif);
+            } else {
+                LOGGER.debug("Detected broken attribute modifier entry on item {}.  Attr={}, Modif={}", stack, attr, modif);
+            }
+        });
+        return map;
+    }
+
+    /**
+     * Creates a mutable component starting with the char used to represent a drop-down list.
+     */
+    private static MutableComponent list() {
```

Comment:
This could use a slightly more descriptive method name, given `list` gives the impression it is some sort of `List<...>`

Reply:
Renamed to `listHeader`